pkg/cmd/generateapireference/run.go

```diff
@@ -0,0 +1,270 @@
+	klog.V(1).Infof("%q version %q", cmd.CommandPath(), programversion.Get())
+	cliflag.PrintFlags(cmd.Flags())
+
+	stopCh := signals.StopChannel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	go func() {
+		<-stopCh
+		cancel()
+	}()
+
+	return o.run(ctx, streams)
+}
+
+func (o *GenerateAPIRefsOptions) parseTemplate(tmplFileName string) (*template.Template, error) {
+	tmplFile := filepath.Join(o.TemplatesDir, tmplFileName)
+	t, err := template.New(tmplFileName).Funcs(templateFuncs).ParseFiles(tmplFile)
+	if err != nil {
+		return nil, fmt.Errorf("can't parse template file %q: %w", tmplFile, err)
+	}
+
+	return t, nil
+}
+
+func (o *GenerateAPIRefsOptions) run(ctx context.Context, streams genericclioptions.IOStreams) error {
+	err := os.Mkdir(o.OutputDir, 0777)
+	if err == nil {
+		klog.V(2).InfoS("Created output directory", "Path", o.OutputDir)
+	} else if os.IsExist(err) {
+		klog.V(2).InfoS("Using existing output directory", "Path", o.OutputDir)
+	} else {
+		return fmt.Errorf("can't create output directory %q: %w", o.OutputDir, err)
+	}
+
+	klog.V(1).InfoS("Parsing CRD files")
+
+	groups := map[string][]*ResourceInfo{}
+	for _, crdPath := range o.CustomResourceDefinitionPaths {
+		crdBytes, err := os.ReadFile(crdPath)
+		if err != nil {
+			return fmt.Errorf("can't read crd file %q: %w", crdPath, err)
+		}
+
+		crd := &apiextensionsv1.CustomResourceDefinition{}
+		err = runtime.DecodeInto(
+			Codecs.DecoderToVersion(Codecs.UniversalDeserializer(), apiextensionsv1.SchemeGroupVersion),
+			crdBytes,
+			crd,
+		)
+		if err != nil {
+			return fmt.Errorf("can't decode crd file %q: %w", crdPath, err)
+		}
+
+		for _, version := range crd.Spec.Versions {
+			if version.Schema.OpenAPIV3Schema == nil {
+				return fmt.Errorf("OpenAPIV3Schema is missing")
+			}
+
+			if groups[crd.Spec.Group] == nil {
+				groups[crd.Spec.Group] = []*ResourceInfo{}
+			}
+			gv := metav1.GroupVersion{Group: crd.Spec.Group, Version: version.Name}
+			groups[crd.Spec.Group] = append(groups[crd.Spec.Group], &ResourceInfo{
+				APIVersion:       gv.String(),
+				Group:            gv.Group,
+				Version:          gv.Version,
+				Names:            crd.Spec.Names,
+				Scope:            crd.Spec.Scope,
+				Served:           version.Served,
+				Storage:          version.Storage,
+				Property:         *version.Schema.OpenAPIV3Schema,
+				NestedProperties: IndexNestedProps(version.Schema.OpenAPIV3Schema.Properties),
+			})
+		}
+	}
+
+	if len(groups) == 0 {
+		return fmt.Errorf("no API group found in CRD files")
+	}
+
+	klog.V(1).InfoS("Parsing templates", "Directory", o.TemplatesDir)
+
+	gvIndexTemplate, err := o.parseTemplate(gvIndexTemplateFileName)
+	if err != nil {
+		return err
+	}
+
+	kindTemplate, err := o.parseTemplate(kindTemplateFileName)
+	if err != nil {
+		return err
+	}
+
+	templates := map[string]*template.Template{}
+	for _, tmplFileName := range templateFiles {
+		tmplFile := filepath.Join(o.TemplatesDir, tmplFileName)
```

Comment:
let's use `parseTemplate` func here

Reply:
actually, I forgot to remove this whole block, thx

Reply:
removed

Reply:
right, I didn't notice it wasn't even used anymore